src/homework/sandro/FourInARow/FourInARow.java

```diff
@@ -0,0 +1,305 @@
+package homework.sandro.FourInARow;
+
+import homework.sandro.Colors;
+
+import java.util.Scanner;
+
+public class FourInARow {
+//    static {
+//        if (Math.random() > 0.5) {
+//            throw new RuntimeException("Hi Sandro, was geht ab?");
+//        }
+//    }
+
+    public static void main(String[] args) {
+
+        int[][] board = new int[6][7];
+        int currentplayer = 1;
+        boolean winner = false;
+        boolean empty = true;
+
+        printBoard(board, 6, 7);
+        while (!winner && empty) {                       //Game runs
+            int col;
+            int row;
+            while (true) {                               //Input read + check if legal move and column empty proof
+                col = readInt("pick a Col, Player Number: " + currentplayer);
+                row = emptyColumn(board, col);
+                if (row == -1) {
+                    System.out.println("try again, column is already full");
+                } else {
+                    break;
+                }
+            }
+
+
+            board[row][col] = currentplayer;                //update board with input and print new board
+            printBoard(board, 6, 7);
+
+
+            winner = hasWin(board, currentplayer);          //detect if currentplayer move trigger a win condition
+            empty = hasAnyMoreMoves(board, 6, 7);   //detect if board has empty spots, for Game is Draw detection
+
+            if (currentplayer == 1) {                       // player change, swap to player 2
```

Comment:
Here you have a bug: The loop will continue to run this code even when winner is = true. This means, after player 1 won, player will be swapped to player 2, loop stops and you print player two won. Same behaviour for player 2. So, the one who is printed as winner, is actually the loser.